core/src/tdx.rs

```diff
@@ -0,0 +1,295 @@
+use crate::{platform::SystemTime, AttestationResult, Policy, RatlsError, TeeType};
+use dcap_qvl::quote::{Quote, Report, TDReport10, TDReport15};
+use dcap_qvl::QuoteCollateralV3;
+use hex::{decode, encode};
+use serde::{de::Error as DeError, Deserialize, Deserializer, Serialize, Serializer};
+use sha2::{Digest, Sha256, Sha384};
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct TdxTcbPolicy {
+    #[serde(
+        default,
+        serialize_with = "serialize_hex_opt",
+        deserialize_with = "deserialize_hex_opt"
+    )]
+    pub mrseam: Option<Vec<u8>>,
+    #[serde(
+        default,
+        serialize_with = "serialize_hex_opt",
+        deserialize_with = "deserialize_hex_opt"
+    )]
+    pub mrtmrs: Option<Vec<u8>>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct TcgEvent {
+    #[serde(default, alias = "type", alias = "event")]
+    pub message: Option<String>,
+    #[serde(default, alias = "event_payload")]
+    pub details: Option<String>,
+    #[serde(default)]
+    pub digest: Option<String>,
+    #[serde(default)]
+    pub imr: Option<u32>,
+}
```

Comment:
Why aren't we following the exact format here? To me, all these fields aren't optional (it means less checking later). The type is something different and it's a number. Using different naming could introduce confusion.

```json
{'digest': '344bc51c980ba621aaa00da3ed7436f7d6e549197dfe699515dfa2c6583d95e6412af21c097d473155875ffd561d6790',
  'event': '',
  'event_payload': '2946762858585858585858582d585858582d585858582d585858582d58585858585858585858585829000000c0ff000000000040080000000000',
  'event_type': 2147483658,
  'imr': 0},
```